test/_base_test_case.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+
+from abc import ABC
+import os
+import random
+import torch
+
+
+class BaseTestCase(ABC):
+    def setUp(self):
+        if hasattr(self.__class__, "seed"):
+            seed = self.__class__.seed
+            if os.getenv("UNLOCK_SEED") is None or os.getenv("UNLOCK_SEED").lower() == "false":
+                self.rng_state = torch.get_rng_state()
+                torch.manual_seed(seed)
+                if torch.cuda.is_available():
+                    torch.cuda.manual_seed_all(seed)
+                random.seed(seed)
+
+    def tearDown(self):
+        if hasattr(self, "rng_state"):
+            torch.set_rng_state(self.rng_state)
+
+    def assertAllClose(self, tensor1, tensor2, rtol=1e-4, atol=1e-5, equal_nan=False):
```

Comment:
How does this differ materially from `torch.allclose`?

Reply:
It gives nicer error messages